### PR TITLE
New API functions

### PR DESCRIPTION
**Pull request by carcus88**
Added API functions to hit all the API functions KickBox supports. Can you consider this for the next release? Also fixed what seemed to be a bug in array merge where my options where wiped out.

### CODE REVIEW

**Comment by brandonmwest** [comment]
Thanks for the PR, we will take a look!

**Comment by schaitanya** [comment]
@carcus88 Thanks for the PR. Will review this shortly.

**Comment by schaitanya** [comment]
Pulled in. Made some minor adjustments to accommodate `PUT` in the future. Will release shortly.

**Comment by schaitanya** [comment]
@carcus88 Released in v2.2.3. Thank you

**Comment by danhstevens** [comment]
Thanks @carcus88! Email help -at- kickbox.io with your shirt size and mailing address if you want some free swag ;)

File: lib/Kickbox/HttpClient/HttpClient.php [hunk 1]
```diff
@@ -99,9 +99,13 @@ private function request($path, array $body = [], $httpMethod = 'GET', array $op
             unset($options['headers']);
         }
 
-        $options['body'] = json_encode($body);
+        if($httpMethod == 'PUT') {
+            $options['body'] = join("\n",$body);
```
Comment (carcus88) [inline lib/Kickbox/HttpClient/HttpClient.php]: I agree and that was essentially just to get it to the point where it works well enough. I think if you really want to do it right the PUT request should accept a stream or an already formatted body string.

**Pull request merged**